docs/en/cue/development-guide.md

```diff
@@ -0,0 +1,914 @@
+
+```shell
+a: *1 | int
+b: 3
+c: a & b
+```
+
+Saving it in `third.cue` file.
+
+You can evaluate the result by using `cue eval`:
+
+```shell
+$ cue eval third.cue
+a: 1
+b: 3
+c: 3
+```
+
+* Conditional statement, it's really useful when you have some cascade operations that different value affects different results.
+
+```shell
+price: number
+feel: *"good" | string
+// Feel bad if price is too high
+if price > 100 {
+    feel: "bad"
+}
+price: 200
+```
+
+Saving it in `fourth.cue` file.
+
+You can evaluate the result by using `cue eval`:
+
+```shell
+$ cue eval fourth.cue
+price: 200
+feel:  "bad"
+```
+
+* For Loop: if you want to avoid duplicate, you may want to use for loop. 
+
+```
+#a: {
+	"hello": "Barcelona"
+	"nihao": "Shanghai"
+}
+
+for k, v in #a {
+	"\(k)": {
+		nameLen: len(v)
+		value:   v
+	}
+}
+```
+
+Note that we use `"\( _my-statement_ )"` for inner calculation in string.
+
+For now, you have finished learning all CUE language basic.
+
+## CUE templating and reference
+
+Let's try to define a CUE template with the knowledge just learned.
+
+1. Define a custom CUE type.
+
+```
+#Config: {
+	name:  string
+	value: string
+}
+```
+
+Let's save it in a file called `deployment.cue`.
+
+2. Define a variable named `parameter`, and use the custom CUE type `#Config`. Using `...` in a list means the
+   list can be appended with multiple elements. If we don't add `...`, then `[#Config]` means the list can only have one element in it.
+
+```shell
+parameter: {
+	name: string
+	image: string
+	config: [...#Config]
+}
+```
+
+Append it into the `deployment.cue`.
+
+
+3. Define a more complex `template` variable and reference the variable `parameter`. 
+
+```
+template: {
+	apiVersion: "apps/v1"
+	kind:       "Deployment"
+	spec: {
+		selector: matchLabels: {
+			"app.oam.dev/component": parameter.name
+		}
+		template: {
+			metadata: labels: {
+				"app.oam.dev/component": parameter.name
+			}
+			spec: {
+				containers: [{
+					name:  parameter.name
+					image: parameter.image
+					if parameter["config"] != _|_ {
+						env: parameter.config
+					}
+				}]
+			}}}
+}
+```
+
+People who are familiar with Kubernetes may have understood that is a template of K8s Deployment. The `parameter` part
+is the parameters of the template.
+
+Append it into the `deployment.cue`.
+
+4. Then, let's append the value by:  
+
+```
+parameter:{
+   name: "mytest"
+   image: "nginx:v1"
+   config: [{name:"a",value:"b"}]
+}
+```
+
+5. Finally, let's export it in yaml:
+
+```shell
+$ cue export deployment.cue -e template --out yaml
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: mytest
+        env:
+        - name: a
+          value: b
+        image: nginx:v1
+    metadata:
+      labels:
+        app.oam.dev/component: mytest
+  selector:
+    matchLabels:
+      app.oam.dev/component: mytest
+```
+
+You have learned how to use CUE to write a template, it's really cool, right?
```

Comment:
This is not cool, this is boring platform engineering work. Please always use serious and accurate narrative in platform builder docs. Also, the best doc could only have examples and demos.

Reply:
got